component/site/views/icals/tmpl/export.php

```diff
@@ -138,7 +140,16 @@
 		}
 		// We Need to wrap this according to the specs
 		/* $html .= "DESCRIPTION:".preg_replace("'<[\/\!]*?[^<>]*?>'si","",preg_replace("/\n|\r\n|\r$/","",$a->content()))."\n"; */
-		$html .= $this->setDescription($a->content()) . "\r\n";
+
+		//Check if we should include the link to the event
+		if ($params->get('source_url', 0) == 1) {
+			$link = $a->viewDetailLink($a->yup(),$a->mup(),$a->dup(),true, $params->get('default_itemid', 0));
+			$uri =  JURI::getInstance(JURI::base());
+			$root = $uri->toString(array('scheme', 'host', 'port'));
+			$html .= $this->setDescription($a->content()) . ' ' .JText::_('JEV_EVENT_IMPORTED_FROM') .$root . JRoute::_($link, true, -1) . "\r\n";
```

Comment:
Your extra text needs to be inside setDescription since the method takes care of line wrapping